### PR TITLE
GEODE-8016: publish with non-zero-padded build ids instead of SNAPSHOT.

### PR DESCRIPTION
**Pull request by rhoughton-pivot**
* Artifacts take the form `1.13.0-build.123` instead of `1.13.0-SNAPSHOT`.
* checkPom task has been modified to use a slug instead of an always
changing version.
* Gradle clients will use the greedy "1.13.0-build+" notation
* Maven clients will use semver v1.0 "[1.13.0-build,1.14.0)" notation.

Signed-off-by: Robert Houghton <rhoughton@pivotal.io>

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [x] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [x] Is your initial contribution a single, squashed commit?

- [x] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: ci/scripts/shared_utilities.sh [hunk 1]
```diff
@@ -31,7 +31,7 @@ find-here-test-reports() {
 }
 
 ## Parsing functions for the Concourse Semver resource.
-## These functions expect one input in the form of the resource file, e.g., "1.9.0-SNAPSHOT.325"
+## These functions expect one input in the form of the resource file, e.g., "1.9.0-build.325"
```
Comment (dickcav) [inline ci/scripts/shared_utilities.sh:34]: Can we use 1.14.0 here instead of an old version?

File: gradle.properties [hunk 2]
```diff
@@ -22,13 +22,11 @@
 # The releaseQualifier uses the following conventions:
```
Comment (dickcav) [inline gradle.properties:22]: We don't use these other releaseQualifiers so can we remove them?

File: ci/scripts/execute_build_examples.sh [hunk 3]
```diff
@@ -17,7 +17,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-set -e
+set -ex
```
Comment (onichols-pivotal) [inline ci/scripts/execute_build_examples.sh:20]: just to confirm, was this change intended, or "debugging" that accidentally got left in?

File: ci/scripts/shared_utilities.sh (hunk 1 above)
Comment (rhoughton-pivot) [reply ci/scripts/shared_utilities.sh:34]: Yes. I'll change it during the rebase (rebasing due to conflict).

File: gradle.properties (hunk 2 above)
Comment (rhoughton-pivot) [reply gradle.properties:22]: Can, but I'm not ready to as part of this change.

File: ci/scripts/execute_build_examples.sh (hunk 3 above)
Comment (rhoughton-pivot) [reply ci/scripts/execute_build_examples.sh:20]: It was, and leaks no secrets.
